subsys/net/lib/lwm2m/lwm2m_obj_access_control.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef LWM2M_OBJ_ACCESS_CONTROL_H
+#define LWM2M_OBJ_ACCESS_CONTROL_H
+#include "lwm2m_engine.h"
+#include "lwm2m_object.h"
+
+#define ACCESS_CONTROL_PERM_READ   BIT(0)
+#define ACCESS_CONTROL_PERM_WRITE  BIT(1)
+#define ACCESS_CONTROL_PERM_EXEC   BIT(2)
+#define ACCESS_CONTROL_PERM_DELETE BIT(3)
+#define ACCESS_CONTROL_PERM_CREATE BIT(4)
+
+/**
+ * @brief Creates an access control object instance. Should be called automatically every
+ * time an object instance is created.
+ *
+ * @param obj_id object id of the object instance getting an access control
+ * @param obj_inst_id object instance id of the object instance getting access control
+ * @param server_obj_inst_id object instance id of the server creating the object instance.
+ * If negative, the default value in CONFIG_LWM2M_SERVER_DEFAULT_SSID is used
+ */
+void access_control_add(uint16_t obj_id, uint16_t obj_inst_id, int server_obj_inst_id);
+void access_control_add_obj(uint16_t obj_id, int server_obj_inst_id);
+/* lwm2macces not the above-defined access, but the LWM2M_OP_ ones defined in lwm2m.h */
```

Comment:
I totally don't understand this comment, perhaps seeing a full function documentation would be more helpful.